src/MCPClient/lib/clientScripts/rightsFromCSV.py

```diff
@@ -0,0 +1,381 @@
+
+        # Initialize hash to note which basis/act combinations for file have alredy been imported
+        if not filepath in self.object_basis_act_usage:
+            self.object_basis_act_usage[filepath] = {}
+
+        basis = self.column_value('basis').lower().capitalize()
+
+        if not basis in self.object_basis_act_usage[filepath]:
+            self.object_basis_act_usage[filepath][basis] = {}
+
+        # Check that act is set and normalize value
+        act = self.column_value('grant_act')
+        if not act:
+            self.exit_with_error('No act specified')
+        act = act.lower().capitalize()
+
+        # Process row if basis/act combination for file hasn't yet been imported
+        if not act in self.object_basis_act_usage[filepath][basis]:
+            self.store_row()
+            self.object_basis_act_usage[filepath][basis][act] = True
+        else:
+            print "Skipping duplicate basis/act combination at row", self.rows_processed + 1
+
+        self.rows_processed += 1
+
+    def exit_with_error(self, message, value=None):
+        """Print message to standard error and exit with status 1."""
+        message = '[Row {}] {}'.format(self.rows_processed + 1, message)
+        if value:
+            print >>sys.stderr, message, value
+        else:
+            print >>sys.stderr, message
+        exit(1)
+
+    def store_row(self):
+        """Create rights records in database using row data."""
+        rights_statement = self.generate_rights_statement()
+
+        if rights_statement.rightsbasis == 'Copyright':
+            self.store_copyright_info(rights_statement)
+
+        elif rights_statement.rightsbasis == 'License':
+            self.store_license_info(rights_statement)
+
+        elif rights_statement.rightsbasis == 'Statute':
+            self.store_statute_info(rights_statement)
+
+        elif rights_statement.rightsbasis in ['Other', 'Donor', 'Policy']:
+            self.store_other_info(rights_statement)
+
+        self.store_grant_info(rights_statement)
+
+    def generate_rights_statement(self):
+        """Generate rights statement."""
+        basis = self.column_value('basis').lower().capitalize()
+
+        if basis not in dict(models.RightsStatement.RIGHTS_BASIS_CHOICES):
+            self.exit_with_error('Invalid basis:', basis)
+
+        # Get file data
+        filepath = self.column_value('file')
+        transfer_file = models.File.objects.filter(originallocation='%transferDirectory%' + filepath, transfer_id=self.transfer_uuid).first()
+
+        if (transfer_file is None):
+            self.exit_with_error('Record not found for file', filepath)
+
+        # Create rights statement
+        rights_statement = models.RightsStatement()
+        rights_statement.metadataappliestotype = self.metadata_applies_to_type
+        rights_statement.metadataappliestoidentifier = transfer_file.uuid
+        rights_statement.rightsbasis = basis
+        rights_statement.status = 'ORIGINAL'
```

Comment:
The originally imported rights metadata stays in METS, but new/changed rights.csv are supported during reingest (I checked with Sara and this is how it's supposed to be as per the spec).